ecs.php

```diff
@@ -377,26 +373,13 @@
             ReferenceThrowableOnlySniff::class,
             // The @param, @return, @var and inline @var annotations should keep standard format
             ParamReturnAndVarTagMalformsFixer::class,
-            // Takes `@var` annotation of non-mixed types and adjusts accordingly the property signature.
-            PhpdocToPropertyTypeFixer::class,
-            // Takes `@param` annotations of non-mixed types and adjusts accordingly the function signature.
-            PhpdocToParamTypeFixer::class,
-            // Takes `@return` annotation of non-mixed types and adjusts accordingly the function signature.
-            PhpdocToReturnTypeFixer::class,
-            // Promote constructor properties
-            // @see https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5956
-            RequireConstructorPropertyPromotionSniff::class,
```

Comment:
Why don't we want this? We are already using it everywhere where we have PHP 8 🤔

Reply:
See PR description - we want this, but I think for some teams may be preferred to have the ability to easily split the upgrade into two phases:
1. Upgrade config format but with no big code changes. For this I will release version 4.0.
2. Upgrade code to comply to new fixers. For this I will release version 4.1, immediately after 4.0.

The most agile teams will upgrade directly to 4.1 and get all those fixers. The conservative teams or teams with bigger codebase may upgrade to 4.0 first, and once they are cleared, they can continue with upgrade to 4.1, which may be more difficult, because the version will contain tens of new fixers, see #95 .

Reply:
Oh sorry, missed the description. Yea, it makes sense 👍

Reply:
But wait, I've just realised - those checks are already present in 3.3.x under [ecs-8.0.php](https://github.com/lmc-eu/php-coding-standard/blob/3.3.0/ecs-8.0.php#L66). So teams using PHP 8.0+ with those rules are already using them and if we remove them, it will create only more mess in the codebase. And if not, they can start with using them as step 1 and afterwards just switch to 4.0.0.
I just don't think removing of rules which are already used in some projects is a good idea tbh

Reply:
Yea I know, what I've meant by that is that code, which is not valid in 3.3 with 8.0 rules used and is also will not be valid in 4.1, will be valid in 4.0. So that can create situation where some code will be formated by those rules from the previous version and some not when 4.0 is used.
However, as far as 4.1 will be released with 4.0 at the same time I'm ok with it :)